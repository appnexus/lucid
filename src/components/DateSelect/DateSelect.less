@import (reference) '../../styles/variables.less';
@import (reference) '../../styles/mixins.less';

.lucid-DateSelect {
	display: flex;
	align-items: center;
	font-size: @size-font;

	&-InfiniteSlidePanel {
		flex: 1;
		height: 100%;
	}

	.lucid-DateSelect-chevron {
		user-select: none;
	}

	.lucid-DateSelect-slide {
		position: relative;
		display: flex;
		align-items: stretch;

		.lucid-DateSelect-show-divider& {
			&:after {
				position: absolute;
				top: 0;
				right: 0;
				content: '';
				width: 1px;
				height: 100%;
				background-color: @color-gray-10;
			}

			&.lucid-InfiniteSlidePanel-Slide-in-frame {
				&:after {
					right: -1px;
				}
			}
		}

		.lucid-DateSelect-slide-content {
			margin: @size-XS;
			flex: 1;
			display: flex;
			align-items: stretch;

			.lucid-DateSelect-CalendarMonth {
				flex: 1;
				font-size: 100%;
			}
		}
	}
}
