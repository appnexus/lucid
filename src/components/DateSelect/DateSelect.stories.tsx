import React from 'react';
import createClass from 'create-react-class';
import { Button, DateSelect } from './../../index';
import timemachine from 'timemachine';

export default {
	title: 'Controls/Selectors/DateSelect',
	component: DateSelect,
	parameters: {
		docs: {
			description: {
				component: (DateSelect as any).peek.description,
			},
		},
	},
};

/* Select Date */
export const SelectDate = () => {
	timemachine.config({
		dateString: 'December 25, 2018 13:12:59',
	});

	const Component = createClass({
		getInitialState() {
			return {
				selectedDate: null,
			};
		},

		handleSelectDate(date: any) {
			this.setState({
				selectedDate: date,
			});
		},

		render() {
			const { selectedDate } = this.state;

			return (
				<section style={{ maxWidth: 400 }}>
					<DateSelect
						selectedDays={selectedDate}
						onSelectDate={this.handleSelectDate}
					/>
					selected date:{' '}
					{selectedDate && selectedDate.toLocaleDateString('en-US')}
				</section>
			);
		},
	});

	return <Component />;
};
SelectDate.storyName = 'SelectDate';

/* Select Range */
export const SelectRange = () => {
	timemachine.config({
		dateString: 'December 25, 2018 13:12:59',
	});

	const Component = createClass({
		getInitialState() {
			return {
				selectMode: 'from',
				from: null,
				to: null,
			};
		},

		handleSelectDate(date: any) {
			const { selectMode } = this.state;

			this.setState({
				selectMode: 'to',
				[selectMode]: date,
			});
		},

		handleReset() {
			this.setState({
				selectMode: 'from',
				from: null,
				to: null,
			});
		},

		render() {
			const { selectMode, from, to } = this.state;

			return (
				<section style={{ maxWidth: 400 }}>
					<DateSelect
						from={from}
						to={to}
						selectMode={selectMode}
						onSelectDate={this.handleSelectDate}
					/>

					<div>
						from: {from && from.toLocaleDateString('en-US')}, to:{' '}
						{to && to.toLocaleDateString('en-US')}
					</div>

					<Button onClick={this.handleReset}>Reset</Button>
				</section>
			);
		},
	});

	return <Component />;
};
SelectRange.storyName = 'SelectRange';

/* Disabled Days */
export const DisabledDays = () => {
	timemachine.config({
		dateString: 'December 25, 2018 13:12:59',
	});

	function isPastDay(date: any) {
		const day = new Date(date.getTime()).setHours(0, 0, 0, 0);
		const today = new Date().setHours(0, 0, 0, 0);
		return day < today;
	}

	const Component = createClass({
		render() {
			return (
				<section style={{ maxWidth: 400 }}>
					<DateSelect disabledDays={isPastDay} />
				</section>
			);
		},
	});

	return <Component />;
};
DisabledDays.storyName = 'DisabledDays';

/* Initial Month */
export const InitialMonth = () => {
	timemachine.config({
		dateString: 'December 25, 2018 13:12:59',
	});

	const Component = createClass({
		render() {
			return (
				<section style={{ maxWidth: 400 }}>
					<DateSelect
						initialMonth={new Date(2016, 1)}
						selectedDays={new Date(2016, 1, 17) as any}
					/>
				</section>
			);
		},
	});

	return <Component />;
};
InitialMonth.storyName = 'InitialMonth';

/* Mutiple Months */
export const MutipleMonths = () => {
	timemachine.config({
		dateString: 'December 25, 2018 13:12:59',
	});

	const Component = createClass({
		render() {
			return (
				<DateSelect
					selectedDays={new Date() as any}
					monthsShown={3}
					calendarsRendered={9}
					showDivider
				/>
			);
		},
	});

	return <Component />;
};
MutipleMonths.storyName = 'MutipleMonths';

/* Custom Modifiers */
export const CustomModifiers = () => {
	timemachine.config({
		dateString: 'December 25, 2018 13:12:59',
	});

	const Component = createClass({
		render() {
			return (
				<section style={{ maxWidth: 400 }}>
					<DateSelect>
						<DateSelect.CalendarMonth
							modifiers={
								{
									tuesday: (date: any) => date.getDay() === 2,
								} as any
							}
						/>
					</DateSelect>

					<style>{`
					.lucid-CalendarMonth .DayPicker-Day--tuesday {
						border: 1px dotted gray;
					}
				`}</style>
				</section>
			);
		},
	});

	return <Component />;
};
CustomModifiers.storyName = 'CustomModifiers';

/* Relative Font Size */
export const RelativeFontSize = () => {
	timemachine.config({
		dateString: 'December 25, 2018 13:12:59',
	});

	const Component = createClass({
		getInitialState() {
			return {
				selectedDate: null,
			};
		},

		handleSelectDate(date: any) {
			this.setState({
				selectedDate: date,
			});
		},

		render() {
			const { selectedDate } = this.state;

			return (
				<section>
					<DateSelect
						isFontSizeRelative
						style={{
							width: 600,
							height: 400,
						}}
						selectedDays={selectedDate}
						onSelectDate={this.handleSelectDate}
					/>
					selected date:{' '}
					{selectedDate && selectedDate.toLocaleDateString('en-US')}
				</section>
			);
		},
	});

	return <Component />;
};
RelativeFontSize.storyName = 'RelativeFontSize';

/* Disable Slide Panel */
export const DisableSlidePanel = () => {
	timemachine.config({
		dateString: 'December 25, 2018 13:12:59',
	});

	const Component = createClass({
		getInitialState() {
			return {
				selectedDate: null,
			};
		},

		handleSelectDate(date: any) {
			this.setState({
				selectedDate: date,
			});
		},

		render() {
			const { selectedDate } = this.state;

			return (
				<section style={{ maxWidth: 800 }}>
					<DateSelect
						useSlidePanel={false}
						selectedDays={selectedDate}
						onSelectDate={this.handleSelectDate}
						monthsShown={2}
						showDivider
					/>
					selected date:{' '}
					{selectedDate && selectedDate.toLocaleDateString('en-US')}
				</section>
			);
		},
	});

	return <Component />;
};
DisableSlidePanel.storyName = 'DisableSlidePanel';
