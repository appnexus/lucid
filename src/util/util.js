import _ from 'lodash';
import React from 'react';

export const isDevMode = (function isReactInDev() {
	try {
		React.createClass({});
	} catch(e) {
		if (e.message.indexOf('render') !== 0) {
			return true;  // A nice, specific error message
		}
	}

	return false;  // should never happen, but play it safe.
})();

export const isNode = typeof process === 'object' && process.title === 'node';

export const logger = (function() {
	return checkIsDev() ? {
		log,
		warn,
		error
	} : {
		log: _.noop,
		warn: _.noop,
		error: _.noop
	};
})();

export function checkIsDev() {
	return isDevMode &&
		typeof window !== 'undefined' &&
		!isNode &&
		typeof console !== 'undefined';
}

function log(...args) {

	console.log(...args);

	try {
		// --- Welcome to debugging Lucid ---
		// This error was thrown as a convenience so that you can use this
		// stack to find the callsite that caused this warning to fire.
		throw new Error(args[0]);
	} catch (x) {}

}

function warn(...args) {

	console.warn(...args);

	try {
		// --- Welcome to debugging Lucid ---
		// This error was thrown as a convenience so that you can use this
		// stack to find the callsite that caused this warning to fire.
		throw new Error(args[0]);
	} catch (x) {}

}

function error(...args) {

	console.error(...args);

	try {
		// --- Welcome to debugging Lucid ---
		// This error was thrown as a convenience so that you can use this
		// stack to find the callsite that caused this warning to fire.
		throw new Error(args[0]);
	} catch (x) {}

}
