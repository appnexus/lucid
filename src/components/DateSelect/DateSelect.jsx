import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import ReactDOM from 'react-dom';
import { buildHybridComponent } from '../../util/state-management';
import { lucidClassNames } from '../../util/style-helpers';
import {
	createClass as createReactClass,
	getFirst,
	omitProps,
} from '../../util/component-types';
import * as reducers from './DateSelect.reducers';
import InfiniteSlidePanel from '../InfiniteSlidePanel/InfiniteSlidePanel';
import CalendarMonth from '../CalendarMonth/CalendarMonth';
import ChevronThinIcon from '../Icon/ChevronThinIcon/ChevronThinIcon';
import { DateUtils } from 'react-day-picker';

const cx = lucidClassNames.bind('&-DateSelect');

const NAV_BUTTON_SIZE = 32;
const clampMonthsShown = monthsShown => _.clamp(monthsShown, 1, 6);

const { any, bool, func, instanceOf, number, oneOf, string } = PropTypes;

/**
 * {"categories": ["controls", "selectors"], "madeFrom": ["InfiniteSlidePanel", "CalendarMonth"]}
 *
 * Date selection component capabaple of supporting single date and date range
 * selections.
 */
const DateSelect = createReactClass({
	displayName: 'DateSelect',

	components: {
		/**
		 * Child component to pass thru props to underlying CalendarMonth.
		 */
		CalendarMonth: createReactClass({
			displayName: 'DateSelect.CalendarMonth',
			propName: 'CalendarMonth',
		}),
	},

	propTypes: {
		/**
		 * Appended to the component-specific class names set on the root element.
		 */
		className: string,

		/**
		 * Number of calendar months to show. Min 1, suggested max 3. Actual max is
		 * 6.
		 */
		monthsShown: number,

		/**
		 * Number of calendar months rendered at any given time (including those
		 * out of view).
		 *
		 * In practice it should be at least (2 * monthsShown) + 2. It's got some
		 * issues that still need to be ironed out but it works.
		 */
		calendarsRendered: number,

		/**
		 * The offset of the leftmost month in view, where 0 is the `initialMonth`.
		 * Negative values will show previous months.
		 */
		offset: number,

		/**
		 * Sets the start date in a date range.
		 */
		from: instanceOf(Date),

		/**
		 * Sets the end date in a date range.
		 */
		to: instanceOf(Date),

		/**
		 * The next selection that is expected. Primarily used to preview expected
		 * ranges when the cursor is on a target date.
		 */
		selectMode: oneOf(['day', 'from', 'to']),

		/**
		 * Sets first month in view on render. The 0 value for the `offset` prop
		 * refers to this month.
		 */
		initialMonth: instanceOf(Date),

		/**
		 * Sets selected days. Passed through to `CalendarMonth` ->
		 * `react-day-picker`. Can be a `Date`, array of `Date`s or a function with
		 * the signature `(date) => Boolean`.
		 */
		selectedDays: any,

		/**
		 * Sets disabled days. Passed through to `CalendarMonth` ->
		 * `react-day-picker`. Can be a `Date`, array of `Date`s or a function with
		 * the signature `(date) => Boolean`.
		 */
		disabledDays: any,

		/**
		 * Display a divider between each month.
		 */
		showDivider: bool,

		/**
		 * Called when user's swipe would change the month `offset`. Callback
		 * passes number of months swiped by the user (positive for forward swipes,
		 * negative for backwards swipes).
		 *
		 * Signature: `(monthsSwiped, { event, props }) => {}`
		 */
		onSwipe: func,

		/**
		 * Called when user clicks the previous button.
		 *
		 * Signature: `({ event, props }) => {}`
		 */
		onPrev: func,

		/**
		 * Called when user clicks the next button.
		 *
		 * Signature: `({ event, props }) => {}`
		 */
		onNext: func,

		/**
		 * Called when user selects a date. Callback passes a Date object as the
		 * first argument.
		 *
		 * Signature: `(selectedDate, { event, props }) => {}`
		 */
		onSelectDate: func,

		/**
		 * Render initial font size relative to size of the component so it scales with the calendar size.
		 */
		isFontSizeRelative: bool,

		/**
		 * Highlight dates and ranges based on cursor position.
		 */
		showCursorHighlight: bool,
	},

	getDefaultProps() {
		return {
			monthsShown: 1,
			calendarsRendered: 6,
			offset: 0,
			from: null,
			to: null,
			initialMonth: new Date(),
			selectedDays: null,
			disabledDays: null,
			showDivider: false,
			onSwipe: _.noop,
			onPrev: _.noop,
			onNext: _.noop,
			onSelectDate: _.noop,
			isFontSizeRelative: false,
			showCursorHighlight: true,
		};
	},

	reducers,

	getInitialState() {
		return {
			cursor: null,
		};
	},

	handleDayClick(day, { disabled }, event) {
		const { onSelectDate } = this.props;

		if (!disabled) {
			onSelectDate(day, { event, props: this.props });
		}
	},

	handleDayMouseEnter(day, { disabled }) {
		if (disabled) {
			this.setState({
				cursor: null,
			});
		} else {
			this.setState({
				cursor: day,
			});
		}
	},

	handleDayMouseLeave() {
		this.setState({
			cursor: null,
		});
	},

	handlePrev(event) {
		this.props.onPrev({ event, props: this.props });
	},

	handleNext(event) {
		this.props.onNext({ event, props: this.props });
	},

	componentWillMount() {
		this.initialMonth = new Date(this.props.initialMonth);
	},

	componentDidMount() {
		const { isFontSizeRelative, monthsShown: monthsShownRaw } = this.props;

		const monthsShown = clampMonthsShown(monthsShownRaw);

		if (isFontSizeRelative) {
			const rootElement = ReactDOM.findDOMNode(this.rootRef);
			const { width, height } = rootElement.getBoundingClientRect();
			const navButtonsWidth = NAV_BUTTON_SIZE * 2;
			const oneMonthShownWidth =
				(width - navButtonsWidth) / monthsShown + navButtonsWidth;
			const size = Math.sqrt(oneMonthShownWidth * height);
			const relativeFontSize = Math.round(size / 24);
			const relativeMinWidth =
				(width - navButtonsWidth) /
					monthsShown *
					10.1075 /
					relativeFontSize *
					monthsShown +
				navButtonsWidth;

			rootElement.style.fontSize = `${relativeFontSize}px`;
			rootElement.style.minWidth = `${relativeMinWidth}px`;
		}
	},

	render() {
		const {
			className,
			monthsShown: monthsShownRaw,
			calendarsRendered,
			offset,
			from,
			to,
			selectMode,
			selectedDays,
			disabledDays,
			showDivider,
			onSwipe,
			showCursorHighlight,
			...passThroughs
		} = this.props;

		const { cursor } = this.state;
		const isRangeSameDay = DateUtils.isSameDay(from, to);

		const calendarMonth = getFirst(
			this.props,
			DateSelect.CalendarMonth,
			<DateSelect.CalendarMonth />
		);
		const monthsShown = clampMonthsShown(monthsShownRaw);

		/* istanbul ignore next */
		return (
			<section
				ref={ref => {
					this.rootRef = ref;
				}}
				className={cx('&', className, {
					'&-show-divider': showDivider,
				})}
				style={{
					minWidth: NAV_BUTTON_SIZE * 2 + 185 * monthsShown,
					...passThroughs.style,
				}}
				{...omitProps(passThroughs, DateSelect)}
			>
				<div>
					<ChevronThinIcon
						className={cx('&-chevron')}
						size={NAV_BUTTON_SIZE}
						isClickable
						direction="left"
						onClick={this.handlePrev}
					/>
				</div>
				<InfiniteSlidePanel
					className={cx('&-InfiniteSlidePanel')}
					totalSlides={calendarsRendered}
					slidesToShow={monthsShown}
					offset={offset}
					onSwipe={onSwipe}
				>
					<InfiniteSlidePanel.Slide className={cx('&-slide')}>
						{slideOffset => (
							<div className={cx('&-slide-content')}>
								<CalendarMonth
									className={cx('&-CalendarMonth')}
									monthOffset={slideOffset}
									initialMonth={this.initialMonth}
									cursor={cursor}
									from={isRangeSameDay ? null : from}
									to={isRangeSameDay ? null : to}
									selectedDays={isRangeSameDay ? from : selectedDays}
									disabledDays={disabledDays}
									selectMode={selectMode}
									onDayClick={this.handleDayClick}
									onDayMouseEnter={
										showCursorHighlight ? this.handleDayMouseEnter : null
									}
									onDayMouseLeave={
										showCursorHighlight ? this.handleDayMouseLeave : null
									}
									// Only update CalendarMonths within frame or one position
									// away. This has issues and doesnt always behave quite right
									shouldComponentUpdate={
										slideOffset - offset >= -1 &&
											slideOffset - offset < monthsShown + 1
									}
									{...calendarMonth.props}
								/>
							</div>
						)}
					</InfiniteSlidePanel.Slide>
				</InfiniteSlidePanel>
				<div>
					<ChevronThinIcon
						className={cx('&-chevron')}
						size={NAV_BUTTON_SIZE}
						isClickable
						direction="right"
						onClick={this.handleNext}
					/>
				</div>
			</section>
		);
	},
});

export default buildHybridComponent(DateSelect);
export { DateSelect as DateSelectDumb };
