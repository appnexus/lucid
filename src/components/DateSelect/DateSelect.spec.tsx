import { shallow } from 'enzyme';
import React from 'react';
import { common, mockDate } from '../../util/generic-tests';
import { DateSelectDumb as DateSelect } from './DateSelect';

describe('DateSelect', () => {
	describe('common tests', () => {
		beforeEach(() => {
			mockDate('2016-02-17T14:25:23-0800');
		});

		afterEach(() => {
			mockDate.restore();
		});

		common(DateSelect);
	});

	describe('event handlers', () => {
		it('should trigger `onSelectDate` when a non-disabled day is clicked', () => {
			const onSelectDate = jest.fn();
			const mockEvent = {};
			const selectDate = new Date('2017-02-28T00:00:00Z');
			const wrapper = shallow(<DateSelect onSelectDate={onSelectDate} />);
			const dateSelectInstance = wrapper.instance();
			dateSelectInstance.handleDayClick(
				selectDate,
				{ disabled: false },
				mockEvent
			);

			expect(onSelectDate).toHaveBeenCalledWith(selectDate, {
				event: mockEvent,
				props: dateSelectInstance.props,
			});
		});

		it('should not trigger `onSelectDate` when a disabled day is clicked', () => {
			const onSelectDate = jest.fn();
			const mockEvent = {};
			const selectDate = new Date('2017-02-28T00:00:00Z');
			const wrapper = shallow(<DateSelect onSelectDate={onSelectDate} />);
			const dateSelectInstance = wrapper.instance();
			dateSelectInstance.handleDayClick(
				selectDate,
				{ disabled: true },
				mockEvent
			);

			expect(onSelectDate).not.toHaveBeenCalled();
		});

		it('should trigger onPrev handler', () => {
			const onPrev = jest.fn();
			const mockEvent = {};
			const wrapper = shallow(
				<DateSelect
					initialMonth={new Date('2017-02-00T00:00:00Z')}
					onPrev={onPrev}
				/>
			);
			const dateSelectInstance = wrapper.instance();
			dateSelectInstance.handlePrev({ event: mockEvent });

			expect(onPrev).toHaveBeenCalledWith({
				event: mockEvent,
				props: dateSelectInstance.props,
			});
		});

		it('should trigger onNext handler', () => {
			const onNext = jest.fn();
			const mockEvent = {};
			const wrapper = shallow(
				<DateSelect
					initialMonth={new Date('2017-02-00T00:00:00Z')}
					onNext={onNext}
				/>
			);
			const dateSelectInstance = wrapper.instance();
			dateSelectInstance.handleNext({ event: mockEvent });

			expect(onNext).toHaveBeenCalledWith({
				event: mockEvent,
				props: dateSelectInstance.props,
			});
		});
	});

	describe('handling cursor state events', () => {
		it('should set cursor state on handleMouseEnter for non-disabled date', () => {
			const cursorDate = new Date('2017-02-17T00:00:00Z');
			const wrapper = shallow(
				<DateSelect initialMonth={new Date('2017-02-01T00:00:00Z')} />
			);
			const dateSelectInstance = wrapper.instance();
			dateSelectInstance.handleDayMouseEnter(cursorDate, { disabled: false });
			expect(dateSelectInstance.state.cursor).toBe(cursorDate);
		});

		it('should not set cursor state on handleMouseEnter for disabled date', () => {
			const cursorDate = new Date('2017-02-17T00:00:00Z');
			const wrapper = shallow(
				<DateSelect initialMonth={new Date('2017-02-01T00:00:00Z')} />
			);
			const dateSelectInstance = wrapper.instance();
			dateSelectInstance.handleDayMouseEnter(cursorDate, { disabled: true });
			expect(dateSelectInstance.state.cursor).not.toBe(cursorDate);
		});

		it('should unset cursor state on handleMouseLeave', done => {
			const cursorDate = new Date('2017-02-17T00:00:00Z');
			const wrapper = shallow(
				<DateSelect initialMonth={new Date('2017-02-01T00:00:00Z')} />
			);
			const dateSelectInstance = wrapper.instance();
			wrapper.setState(
				{
					cursor: cursorDate,
				},
				() => {
					dateSelectInstance.handleDayMouseLeave();
					wrapper.setState({}, () => {
						expect(dateSelectInstance.state.cursor).toBeNull();
						done();
					});
				}
			);
		});
	});
});
