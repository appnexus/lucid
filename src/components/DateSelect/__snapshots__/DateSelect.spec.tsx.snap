// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSelect common tests [common] example testing should match snapshot(s) for 1.select-date 1`] = `
<section
  className="lucid-DateSelect"
  style={
    Object {
      "minWidth": 249,
    }
  }
>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="left"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
  <InfiniteSlidePanel
    className="lucid-DateSelect-InfiniteSlidePanel lucid-DateSelect-slidePanel"
    isAnimated={true}
    isLooped={false}
    offset={0}
    onSwipe={[Function]}
    slidesToShow={1}
    totalSlides={6}
  >
    <InfiniteSlidePanel.Slide
      className="lucid-DateSelect-slide"
    >
      <Component />
    </InfiniteSlidePanel.Slide>
  </InfiniteSlidePanel>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="right"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
</section>
`;

exports[`DateSelect common tests [common] example testing should match snapshot(s) for 2.select-range 1`] = `
<section
  className="lucid-DateSelect"
  style={
    Object {
      "minWidth": 249,
    }
  }
>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="left"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
  <InfiniteSlidePanel
    className="lucid-DateSelect-InfiniteSlidePanel lucid-DateSelect-slidePanel"
    isAnimated={true}
    isLooped={false}
    offset={0}
    onSwipe={[Function]}
    slidesToShow={1}
    totalSlides={6}
  >
    <InfiniteSlidePanel.Slide
      className="lucid-DateSelect-slide"
    >
      <Component />
    </InfiniteSlidePanel.Slide>
  </InfiniteSlidePanel>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="right"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
</section>
`;

exports[`DateSelect common tests [common] example testing should match snapshot(s) for 3.disabled-days 1`] = `
<section
  className="lucid-DateSelect"
  style={
    Object {
      "minWidth": 249,
    }
  }
>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="left"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
  <InfiniteSlidePanel
    className="lucid-DateSelect-InfiniteSlidePanel lucid-DateSelect-slidePanel"
    isAnimated={true}
    isLooped={false}
    offset={0}
    onSwipe={[Function]}
    slidesToShow={1}
    totalSlides={6}
  >
    <InfiniteSlidePanel.Slide
      className="lucid-DateSelect-slide"
    >
      <Component />
    </InfiniteSlidePanel.Slide>
  </InfiniteSlidePanel>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="right"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
</section>
`;

exports[`DateSelect common tests [common] example testing should match snapshot(s) for 4.initial-month 1`] = `
<section
  className="lucid-DateSelect"
  style={
    Object {
      "minWidth": 249,
    }
  }
>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="left"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
  <InfiniteSlidePanel
    className="lucid-DateSelect-InfiniteSlidePanel lucid-DateSelect-slidePanel"
    isAnimated={true}
    isLooped={false}
    offset={0}
    onSwipe={[Function]}
    slidesToShow={1}
    totalSlides={6}
  >
    <InfiniteSlidePanel.Slide
      className="lucid-DateSelect-slide"
    >
      <Component />
    </InfiniteSlidePanel.Slide>
  </InfiniteSlidePanel>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="right"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
</section>
`;

exports[`DateSelect common tests [common] example testing should match snapshot(s) for 5.mutiple-months 1`] = `
<section
  className="lucid-DateSelect lucid-DateSelect-show-divider"
  style={
    Object {
      "minWidth": 619,
    }
  }
>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="left"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
  <InfiniteSlidePanel
    className="lucid-DateSelect-InfiniteSlidePanel lucid-DateSelect-slidePanel"
    isAnimated={true}
    isLooped={false}
    offset={0}
    onSwipe={[Function]}
    slidesToShow={3}
    totalSlides={9}
  >
    <InfiniteSlidePanel.Slide
      className="lucid-DateSelect-slide"
    >
      <Component />
    </InfiniteSlidePanel.Slide>
  </InfiniteSlidePanel>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="right"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
</section>
`;

exports[`DateSelect common tests [common] example testing should match snapshot(s) for 6.custom-modifiers 1`] = `
<section
  className="lucid-DateSelect"
  style={
    Object {
      "minWidth": 249,
    }
  }
>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="left"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
  <InfiniteSlidePanel
    className="lucid-DateSelect-InfiniteSlidePanel lucid-DateSelect-slidePanel"
    isAnimated={true}
    isLooped={false}
    offset={0}
    onSwipe={[Function]}
    slidesToShow={1}
    totalSlides={6}
  >
    <InfiniteSlidePanel.Slide
      className="lucid-DateSelect-slide"
    >
      <Component />
    </InfiniteSlidePanel.Slide>
  </InfiniteSlidePanel>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="right"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
</section>
`;

exports[`DateSelect common tests [common] example testing should match snapshot(s) for 7.relative-font-size 1`] = `
<section
  className="lucid-DateSelect"
  style={
    Object {
      "height": 400,
      "width": 600,
    }
  }
>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="left"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
  <InfiniteSlidePanel
    className="lucid-DateSelect-InfiniteSlidePanel lucid-DateSelect-slidePanel"
    isAnimated={true}
    isLooped={false}
    offset={0}
    onSwipe={[Function]}
    slidesToShow={1}
    totalSlides={6}
  >
    <InfiniteSlidePanel.Slide
      className="lucid-DateSelect-slide"
    >
      <Component />
    </InfiniteSlidePanel.Slide>
  </InfiniteSlidePanel>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="right"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
</section>
`;

exports[`DateSelect common tests [common] example testing should match snapshot(s) for 8.disable-slide-panel 1`] = `
<section
  className="lucid-DateSelect lucid-DateSelect-show-divider"
  style={
    Object {
      "minWidth": 434,
    }
  }
>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="left"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
  <div
    className="lucid-DateSelect-slidePanel lucid-DateSelect-slidePanel-simple"
  >
    <div
      className="lucid-DateSelect-slide lucid-DateSelect-slide-simple"
      key="0"
    >
      <div
        className="lucid-DateSelect-slide-content"
      >
        <CalendarMonth
          className="lucid-DateSelect-CalendarMonth"
          cursor={null}
          disabledDays={[Function]}
          from={null}
          initialMonth={2016-02-17T22:25:23.000Z}
          monthOffset={0}
          onDayClick={[Function]}
          onDayMouseEnter={[Function]}
          onDayMouseLeave={[Function]}
          selectMode="day"
          selectedDays={null}
          shouldComponentUpdate={true}
          to={null}
        />
      </div>
    </div>
    <div
      className="lucid-DateSelect-slide lucid-DateSelect-slide-simple"
      key="1"
    >
      <div
        className="lucid-DateSelect-slide-content"
      >
        <CalendarMonth
          className="lucid-DateSelect-CalendarMonth"
          cursor={null}
          disabledDays={[Function]}
          from={null}
          initialMonth={2016-02-17T22:25:23.000Z}
          monthOffset={1}
          onDayClick={[Function]}
          onDayMouseEnter={[Function]}
          onDayMouseLeave={[Function]}
          selectMode="day"
          selectedDays={null}
          shouldComponentUpdate={true}
          to={null}
        />
      </div>
    </div>
  </div>
  <div>
    <ChevronIcon
      aspectRatio="xMidYMid meet"
      className="lucid-DateSelect-chevron"
      color="primary"
      direction="right"
      isClickable={true}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={32}
      viewBox="0 0 16 16"
    />
  </div>
</section>
`;
